src/libstore/store-api.cc

```diff
@@ -394,7 +394,7 @@ OutputPathMap Store::queryDerivationOutputMap(const StorePath & path) {
     OutputPathMap result;
     for (auto & [outName, optOutPath] : resp) {
         if (!optOutPath)
-            throw Error("output '%s' has no store path mapped to it", outName);
+            throw Error("output '%s' of derivation '%s' has no store path mapped to it", outName, path.to_string());
```

Comment:
Shouldn't this be `printStorePath(path)`?

Reply:
Oh indeed. Fixed